File: internal/types/features.go
package types //nolint: revive,nolintlint

type RawFeaturesFile struct {
	Type     string `json:"type"`
	Features []struct {
		Type       string                 `json:"type"`
		Properties map[string]interface{} `json:"properties"`
	} `json:"features"`
}

type Features struct {
	CachedObject

	Class   string         `json:"class"`
	Count   int            `json:"featuresCount"`
	Objects map[string]int `json:"objects"`
}

File: frontend/src/models/features.ts
import type { CachedObject } from "@/models/common";

export class Features {
  class: string;
  count: number;
  objects: Record<string, number>;
  cachedObject: CachedObject;

  constructor(
    clazz: string,
    count: number,
    objects: Record<string, number>,
    cachedObject: CachedObject
  ) {
    this.class = clazz;
    this.count = count;
    this.objects = objects;
    this.cachedObject = cachedObject;
  }
}
